### ISSUE TITLE
Enforce Intune MAM App Protection Policy

### ISSUE THREAD

**Issue by andrejacobs** [msintuneappsdk/ms-intune-app-sdk-ios#46]
Hi,

Our Intune MAM enabled app used to have MAMPolicyRequired and AutoEnrollOnLaunch set to True/Yes and if I recall correctly, the app will be blocked by Intune MAM screens until a MAM policy was installed. I.e. the app would not be allowed to continue to run further.

However because our app is very dynamic and driven based on server / environment configurations and we make use of ADAL (depending on client configurations) the Intune MAM enabled way of enrolling had to be changed (so the user is not double prompted for login etc.) and thus we can now no longer rely on Info.plist settings. Thus at runtime I decide which one of the 2 enrollment api calls to make.

According to the documentation, the IntuneMAMEnrollmentDelegate should only be used for logging / debugging and not tied to business logic at all.

So my question is:
* How would you recommend that I "block" our app from logging in the user / showing him content until we know that an Intune MAM policy is inplace?

At the moment there is a chance where the enrollment fails, the app loads further and the user is able to overcome security (for example being able to copy and paste from the app).

Kind regards,
Andre

**Comment by Kyle-Reis** [comment]
Hey @andrejacobs, which documentation are you referring to?

**Comment by andrejacobs** [comment]
@Kyle-Reis [app-sdk-ios#status-result-and-debug-notifications](https://docs.microsoft.com/en-us/intune/app-sdk-ios#status-result-and-debug-notifications)

Essentially what I require is the ability to tell at runtime if the app/user has been successfully enrolled into Intune MAM and the SDK has an app protection policy in place such that the user can then be allowed to use the app. If there is not policy in place then the user will be able to overcome security measurements like for example copy and paste confidential information.

Also a bit higher up in the documentation it states:

> By calling the registerAndEnrollAccount method, the SDK will register the user account and attempt to enroll the app on behalf of this account. If the enrollment fails for any reason, the SDK will automatically retry the enrollment 24 hours later. For debugging purposes, the app can receive notifications, via a delegate, about the results of any enrollment requests.

**Comment by andrejacobs** [comment]
So the approach I am trying at the moment is.
If we have no policy (PolicyManager.policy) then
- Show a VC with an activity indicator informing the user we are enrolling them into Intune MAM.
- Enroll using one of the two methods (depends whether ADAL is being used or not).
- Business logic in the Delegate, if a success, then dismiss the VC else, show an error message to the user and ask them to quit the app and retry.

In other words this goes against the Intune MAM SDK documentation saying not to put business logic in the delegate calls, but this does follow the Wagr app example (which controls business logic for the button).

@Kyle-Reis Does this sound like a reasonable approach to ensure the user can not progress with the app unless a MAM App Protection Policy is in place?

**Comment by Kyle-Reis** [comment]
Hey @andrejacobs, this is a perfectly valid scenario. Particularly for LOB apps that are developed specifically for Intune-managed users. I think that note was geared toward our own 1st party partners and 3rd party public store apps who may have AAD users that are not Intune licensed, etc. As you can see above, I've submitted a PR to remove that note from the doc.

A couple of scenarios you may want to consider:
1.) Users can be licensed for Intune and enrolled, but may not have policy assigned to them. This could potentially be intentional. Depending on the policy of your organization (whether or not this is allowed), you may want to allow the user to access corporate data in this case, or you may choose to show a more descriptive message so the user knows to contact the IT department and get policy assigned to them.
2.) At any point after enrolling, the SDK could check in with the MAM service and find that policy is no longer targeted. Again, depending on whether or not this is allowed in your organization, you may choose to handle this differently.

**Comment by andrejacobs** [comment]
@Kyle-Reis Thanks for the clarification and I will look into the scenarios you mentioned.
